addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -181,6 +181,11 @@ def _cart_update(self, product_id, line_id=None, add_qty=0, set_qty=0, **kwargs)
         self._auto_apply_rewards()
         return res
 
+    def _get_non_delivery_lines(self):
+        """Override of `website_sale` to exclude delivery reward lines."""
+        res = super()._get_non_delivery_lines()
+        return res.filtered(lambda l: l.reward_id.reward_type != 'shipping')
+
```

Comment:
Why not use `_get_free_shipping_lines`?
```suggestion
        return super()._get_non_delivery_lines() - self._get_free_shipping_lines()
```